config/value.go

```diff
@@ -337,11 +338,17 @@ func convertValue(value interface{}, targetType reflect.Type) (interface{}, erro
 	}
 	switch v := value.(type) {
 	case string:
-		switch targetType.Name() {
-		case "int":
+		target := reflect.New(targetType).Interface()
+		switch target.(type) {
+		case *int:
 			return strconv.Atoi(v)
-		case "bool":
-			return v == "True" || v == "true", nil
+		case *bool:
+			return strconv.ParseBool(v)
+		case encoding.TextUnmarshaler:
+			err := target.(encoding.TextUnmarshaler).UnmarshalText([]byte(v))
```

Comment:
From my understanding string is just a readonly byte slice, so yes, it will be always happy.